src/SessionSetup/RetrieveInstancesWidget.cpp

```diff
@@ -49,6 +49,17 @@ namespace {
   return project;
 }
 
+void SetQSettingsProject(std::optional<Project> project) {
```

Comment:
Nit: This function name is exposing implementation details. For the caller it's not important that the project is saved using QSettings. But it's important that this persists the project for the next Orbit session. I suggest to call it that way. Something like `SaveProjectToPersistentStorage` / `LoadProjectFromPersistentStorage`. This way we can easily swap out QSettings if needed. (That's a Nit because it's technically not an API surface since it's a static function.)

Reply:
Apologies for bringing this up rather late. `GetQSettingsProject` was already introduced in the last PR.

Reply:
done.